File: _submissions/19UCS159_Tharun/Day5/diffK.java
public class diffK {
  public int diffPossible(int[] A, int B) {
    int n = A.length;
    int i = 0, j = 1;
    while (j < n) {
      int diff = A[j] - A[i];
      if (diff == B)
        return 1;
      if (diff < B)
        j++;
      else {
        i++;
        if (i == j)
          j++;
      }
    }
    return 0;
  }
}

File: _submissions/18UEI056_Prashun/Day5/Diffk.cpp
int Solution::diffPossible(vector<int> &A, int B) {
    if(B>A[A.size()-1]) return 0;
    int i=0,j=1;
    while(i<A.size() && j<A.size())
    {if(A[j]-A[i]==B &&i!=j)
      return 1;
 if(A[j]-A[i]<B)
 j++;
 else
 i++;
    }
   return 0; 
}
